File: CRMConnect.Server/Models/Sale.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace CRMConnect.Server.Models
{
    public class Sale
    {
        public int SaleId { get; set; }
        public string ProductName { get; set; } = string.Empty;
        [Column(TypeName = "decimal(18,2)")]
        public decimal SaleAmount { get; set; }
        public DateTime SaleDate { get; set; }
        public int? SalesRepId { get; set; }
        public SalesRepresentative? SalesRepresentative { get; set; }
        public int? CustomerId { get; set; }
        public Customer? Customer { get; set; }
    }
}

File: crmconnect.client/src/models/Sale.ts
import { Customer } from './Customer';

export interface Sale {
    saleId: number;
    productName: string;
    saleAmount: number;
    saleDate: Date;
    salesRepId?: number;
    salesRepresentative?: SalesRepresentative;
    customerId?: number;
    customer?: Customer;
}
export interface SalesRepresentative {
    salesRepId: number;
    name: string;
    email: string;
}